index.js

```diff
@@ -118,6 +128,30 @@ const meow = (helpText, options) => {
 		}
 	}
 
+	// Get a list of missing flags, that are required
+	const missing = [];
+	for (const requiredFlag of requiredFlags) {
+		let requiredByFunction = true;
+
+		if (typeof options.flags[requiredFlag].isRequired === 'function') {
+			requiredByFunction = options.flags[requiredFlag].isRequired(flags, input);
+		}
+
+		if (typeof flags[requiredFlag] === 'undefined' && requiredByFunction) {
+			missing.push({key: requiredFlag, ...options.flags[requiredFlag]});
+		}
+	}
+
+	// Print error message for missing flags
+	if (missing.length > 0) {
+		console.log(`Missing required option${missing.length > 1 ? 's' : ''}`);
+		for (const flag of missing) {
+			console.log(`\t--${flag.key}${flag.alias ? `, -${flag.alias}` : ''}`);
+		}
+
+		process.exit(2);
+	}
+
```

Comment:
This main function is getting huge. Would be great if you could modularize the logic you're adding into separate methods.